### PR TITLE
Fix + Change to primary constructor

### PR DESCRIPTION
**Pull request by Ceredron**
## Description
Initial commit was a small fix of some code where I had forgotten to set a field variable in the constructor. Quick oversight! But it makes me think it might be worthwhile to start using the [new "Primary Constructor"-syntax introduced in C# 12](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/primary-constructors). Instead of using a constructor, the injected fields are set like so:
```
public class Foo(IFileTransferRepository fileTransferRepository)
{
}
```
With traditional syntax the same would be:
```
public class Foo
{
    private readonly IFileTransferRepository _fileTransferRepository;
    public Foo(IFileTransferRepository fileTransferRepository)
    {
        _fileTransferRepository = fileTransferRepository;
    }
}
```
I think primary constructor is clearer and more concise.

My main gripe with it has been that it cannot always be used. For instance, in [AltinnAuthorizationService.cs](https://github.com/Altinn/altinn-broker/blob/main/src/Altinn.Broker.Integrations/Altinn/Authorization/AltinnAuthorizationService.cs) we call a method in the constructor (httpClient.DefaultRequestHeaders.Add(...)) and thus this class cannot use primary constructor.
However, I see this is very much exception and not rule. Vast majority of our constructors just map parameters to fields.

## Related Issue(s)
- #527 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [x] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Broker.Application/GetFileTransferOverview/GetFileTransferOverviewHandler.cs [hunk 1]
```diff
@@ -9,22 +9,11 @@
 
 namespace Altinn.Broker.Application.GetFileTransferOverview;
 
-public class GetFileTransferOverviewHandler : IHandler<GetFileTransferOverviewRequest, GetFileTransferOverviewResponse>
+public class GetFileTransferOverviewHandler(IAuthorizationService resourceRightsRepository, IFileTransferRepository fileTransferRepository, IResourceManager resourceManager, ILogger<GetFileTransferOverviewHandler> logger) : IHandler<GetFileTransferOverviewRequest, GetFileTransferOverviewResponse>
```
Comment (Andreass2) [inline src/Altinn.Broker.Application/GetFileTransferOverview/GetFileTransferOverviewHandler.cs:12]: Could remove unused

File: src/Altinn.Broker.Application/GetFileTransfers/LegacyGetFilesHandler.cs [hunk 2]
```diff
@@ -8,29 +8,14 @@
 
 namespace Altinn.Broker.Application.GetFileTransfers;
 
-public class LegacyGetFilesHandler : IHandler<LegacyGetFilesRequest, List<Guid>>
+public class LegacyGetFilesHandler(IAuthorizationService resourceRightsRepository, IResourceRepository resourceRepository, IFileTransferRepository fileTransferRepository, IActorRepository actorRepository, ILogger<GetFileTransfersHandler> logger) : IHandler<LegacyGetFilesRequest, List<Guid>>
```
Comment (Andreass2) [inline src/Altinn.Broker.Application/GetFileTransfers/LegacyGetFilesHandler.cs:11]: Could remove unused

File: src/Altinn.Broker.Application/GetFileTransfers/GetFileTransfersHandler.cs [hunk 3]
```diff
@@ -9,36 +9,21 @@
 
 namespace Altinn.Broker.Application.GetFileTransfers;
 
-public class GetFileTransfersHandler : IHandler<GetFileTransfersRequest, List<Guid>>
+public class GetFileTransfersHandler(IAuthorizationService resourceRightsRepository, IResourceRepository resourceRepository, IFileTransferRepository fileTransferRepository, IActorRepository actorRepository, ILogger<GetFileTransfersHandler> logger) : IHandler<GetFileTransfersRequest, List<Guid>>
```
Comment (Andreass2) [inline src/Altinn.Broker.Application/GetFileTransfers/GetFileTransfersHandler.cs:12]: Could remove unused

File: src/Altinn.Broker.API/Controllers/LegacyFileController.cs [hunk 4]
```diff
@@ -146,11 +140,11 @@
         if (recipients?.Length > 0)
         {
             recipientsString = string.Join(',', recipients);
-            _logger.LogInformation("Getting files with status {status} created {from} to {to} for recipients {recipients}", recipientStatus?.ToString(), from?.ToString(), to?.ToString(), recipientsString);
+            logger.LogInformation("Getting files with status {status} created {from} to {to} for recipients {recipients}", recipientStatus?.ToString(), from?.ToString(), to?.ToString(), recipientsString);
```
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.API/Controllers/LegacyFileController.cs:143]: Good bot. This is valid concern.

These strings should already adhere to a format. I will validate them with a Regex first that should filter out injection attempts.

File: src/Altinn.Broker.API/Controllers/LegacyFileController.cs [hunk 5]
```diff
@@ -146,11 +140,11 @@
...
         }
         else
         {
-            _logger.LogInformation("Getting files with status {status} created {from} to {to} for consumer {consumer}", recipientStatus?.ToString(), from?.ToString(), to?.ToString(), onBehalfOfConsumer);
+            logger.LogInformation("Getting files with status {status} created {from} to {to} for consumer {consumer}", recipientStatus?.ToString(), from?.ToString(), to?.ToString(), onBehalfOfConsumer);
```
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.API/Controllers/LegacyFileController.cs:147]: Good bot. This is valid concern.

These strings should already adhere to a format. I will validate them with a Regex first that should filter out injection attempts.

**Review by Andreass2 — APPROVED** [review]
Very nice! Amazing to remove so much boilerplate code!

File: src/Altinn.Broker.Application/GetFileTransfers/GetFileTransfersHandler.cs (hunk 3 above)
Comment (Ceredron) [reply src/Altinn.Broker.Application/GetFileTransfers/GetFileTransfersHandler.cs:12]: Removed

File: src/Altinn.Broker.Application/GetFileTransfers/LegacyGetFilesHandler.cs (hunk 2 above)
Comment (Ceredron) [reply src/Altinn.Broker.Application/GetFileTransfers/LegacyGetFilesHandler.cs]: Removed

**Pull request merged**